openapi-cli/src/main/java/io/ballerina/openapi/generators/schema/SchemaUtils.java

```diff
@@ -0,0 +1,172 @@
+import io.swagger.v3.oas.models.OpenAPI;
+import io.swagger.v3.oas.models.media.Schema;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static io.ballerina.compiler.syntax.tree.AbstractNodeFactory.createEmptyNodeList;
+import static io.ballerina.compiler.syntax.tree.AbstractNodeFactory.createNodeList;
+import static io.ballerina.compiler.syntax.tree.AbstractNodeFactory.createToken;
+import static io.ballerina.compiler.syntax.tree.NodeFactory.createMarkdownDocumentationNode;
+import static io.ballerina.compiler.syntax.tree.NodeFactory.createMetadataNode;
+import static io.ballerina.compiler.syntax.tree.SyntaxKind.SEMICOLON_TOKEN;
+import static io.ballerina.openapi.generators.GeneratorConstants.NILLABLE;
+import static io.ballerina.openapi.generators.GeneratorUtils.escapeIdentifier;
+import static io.ballerina.openapi.generators.GeneratorUtils.getValidName;
+
+/**
+ * Contains util functions needed for schema generation.
+ *
+ * @since 2.0.0
+ */
+public class SchemaUtils {
+    /**
+     * Generate proper type name considering the nullable configurations.
+     * Scenario 1 : schema.getNullable() != null && schema.getNullable() == true && nullable == true -> string?
+     * Scenario 2 : schema.getNullable() != null && schema.getNullable() == true && nullable == false -> string?
+     * Scenario 3 : schema.getNullable() != null && schema.getNullable() == false && nullable == false -> string
+     * Scenario 4 : schema.getNullable() != null && schema.getNullable() == false && nullable == true -> string
+     * Scenario 5 : schema.getNullable() == null && nullable == true -> string?
+     * Scenario 6 : schema.getNullable() == null && nullable == false -> string
+     *
+     * @param schema Schema of the property
+     * @param type   Type name
+     * @return Final type of the field
+     */
+    public static String getNullableType(Schema schema, String type, boolean nullable) {
+        if (schema.getNullable() != null) {
+            if (schema.getNullable()) {
+                type = type + NILLABLE;
+            }
+        } else if (nullable) {
+            type = type + NILLABLE;
+        }
+        return type;
+    }
+
+    /**
+     * This util for generating record field with given schema properties.
+     */
+    public static void addRecordFields(List<String> required, List<Node> recordFieldList, Map.Entry<String,
+            Schema> field, boolean nullable, OpenAPI openAPI)
+            throws BallerinaOpenApiException {
+        // TODO: Handle allOf , oneOf, anyOf
+        RecordFieldNode recordFieldNode;
+        String fieldN = escapeIdentifier(field.getKey().trim());
+        // API doc generations
+        List<Node> schemaDoc = getFieldApiDocs(openAPI, field.getValue());
+        IdentifierToken fieldName = AbstractNodeFactory.createIdentifierToken(fieldN);
+        TypeDescriptorNode fieldTypeName = getSchemaType(field.getValue(), nullable, openAPI);
```

Comment:
WDYT? if we can add function name as `getBallerinaType()` or something related to function behaviour since it has been used for extracting the schema details that map to ballerina types and already defined function is there `getSchemaType` for filter the schema. 🙂

Reply:
Updated to `getBallerinaTypeName`.